### PR TITLE
Add `ContentUserRegistration.isCurrentlyRequired` field

### PR DESCRIPTION
**Pull request by B77Mills**
Add logic to only apply isRequired and access levels when the start and end date for requires Registration is met.

### CODE REVIEW

File: services/graphql-server/src/graphql/resolvers/platform/content.js [hunk 1]
```diff
@@ -535,6 +535,12 @@ module.exports = {
       const requiresRegistrationOptions = getAsObject(content, 'mutations.Website.requiresRegistrationOptions');
       const requiresAccessLevels = get(content, 'mutations.Website.requiresAccessLevels');
       const { startDate, endDate } = requiresRegistrationOptions;
+      if (requiresRegistration && (startDate || endDate)) {
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/resolvers/platform/content.js]: I don't believe you want to change this resolver... if you do, the management interface will no longer function

**Review by zarathustra323 — CHANGES_REQUESTED** [review]

File: services/graphql-server/src/graphql/resolvers/platform/content.js (hunk 1 above)
Comment (zarathustra323) [reply services/graphql-server/src/graphql/resolvers/platform/content.js]: If you're trying to determine if registration is required for gating, that should be handled elsewhere

File: services/graphql-server/src/graphql/resolvers/platform/content.js [hunk 2]
```diff
@@ -310,6 +310,14 @@ module.exports = {
    *
    */
   ContentUserRegistration: {
+    isCurrentlyRequired: async (root) => {
+      if (!root.isRequired) return false;
+      if (!root.startDate && !root.endDate) return root.isRequired;
+      const now = new Date();
+      if (root.startDate && root.startDate <= now) return false;
+      if (root.endDate && root.endDate >= now) return false;
+      return root.isRequired;
```
Comment (solocommand) [inline services/graphql-server/src/graphql/resolvers/platform/content.js]: I think your logic is reversed here; you want to return true if there is a start date and it's in the past. You'd need to check the end date to be sure in that condition though. Something like this?
```suggestion
      if (root.startDate && root.startDate <= now) {
        if (root.endDate && root.endDate <= now) return false;
      }
      return true;
```

**Review by solocommand — DISMISSED** [review]

File: services/graphql-server/src/graphql/resolvers/platform/content.js (hunk 2 above)
Comment (B77Mills) [reply services/graphql-server/src/graphql/resolvers/platform/content.js]: @solocommand I did have the checks wrong, but I think they still are independent conditions.  I changed it to 

```
if (root.startDate && root.startDate > now) return false;
if (root.endDate && root.endDate < now) return false;
```

File: services/graphql-server/src/graphql/resolvers/platform/content.js [hunk 3]
```diff
@@ -310,6 +310,14 @@ module.exports = {
    *
    */
   ContentUserRegistration: {
+    isCurrentlyRequired: async (root) => {
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/resolvers/platform/content.js]: Destructure `root` and update the ifs to use the vars directly...
```js
isCurrentlyRequired: async ({ isRequired, startDate, endDate }) => {
```

File: services/graphql-server/src/graphql/resolvers/platform/content.js [hunk 4]
```diff
@@ -530,11 +538,18 @@ module.exports = {
 
     userRegistration: (content) => {
       const requiresRegistration = get(content, 'mutations.Website.requiresRegistration');
-      if (!requiresRegistration) return { isRequired: false, siteIds: [], accessLevels: [] };
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/resolvers/platform/content.js]: Theses changes can be discarded

**Review by zarathustra323 — CHANGES_REQUESTED** [review]

**Review by zarathustra323 — APPROVED** [review]

**Pull request merged**